packages/CB-serverless-backend/api/order/createOrder.js

```diff
@@ -107,4 +115,66 @@ const getCurrentCart = (userId) => {
 	};
 	
 	return documentClient.get(params).promise();
-}
\ No newline at end of file
+}
+
+const batchUpdateAvailableAndSoldQuantities = (cartItems, revert = false) => {
+  // Get Current Values of the cart for groceryId present in cartItems
+  return getAvailableAndSoldQuantityForGroceries(cartItems)
+    .then(data => Promise.all(
+      map(data.Responses.grocery, (eachGrocery) => {
+        // Finds the index in cart and matches since batch get not guarentees the order
+        const cartIndex = findIndex(cartItems, item => item.groceryId === eachGrocery.groceryId);
+        const orderedQty = cartItems[cartIndex].qty;
+        // Updates the quantity
+        return updateAvailableAndSoldQuantities(eachGrocery, orderedQty, revert);
+      })
+    ))
+    .catch((err) => {
+      console.log(err);
+      return Promise.reject(JSON.stringify(err.message))
+    })
+}
+
+// BatchGet the current sold and available quantities
+const getAvailableAndSoldQuantityForGroceries = (cartItems) => {
+  const keys = map(cartItems, (item) => {
+    return {
+      'groceryId': item.groceryId,
+    }
+  });
+  const params = {
+    RequestItems: {
+      'grocery': {
+        Keys: keys,
+        ProjectionExpression: 'groceryId, availableQty, soldQty'
+      }
+    },
+    
+  }
+  return documentClient.batchGet(params).promise();
+}
+
+// Reverts or cancel reverts of an item - current data
+// By default revert is false which is - It is a placed order (opposite to cancelling)
+const updateAvailableAndSoldQuantities = (currentData, orderedQty, revert = false) => {
+  const factor = revert ? -1 : 1;
+  // Update available and sold qty
+  const updatedAvailableQty = currentData.availableQty - (factor * orderedQty);
+  const updatedSoldQty = currentData.soldQty + (factor * orderedQty);
+
+  const params = {
+    TableName: 'grocery',
```

Comment:
Same here. Please use the name from constants.

Reply:
@Arjun-sna update with the mentioned fixes.